### PR TITLE
fix autoscroll and hopefully jumping too

### PR DESCRIPTION
**Pull request by skylarbarrera**
this comment was the inspo for the jumping fixes: https://github.com/Flipkart/recyclerlistview/issues/581#issuecomment-852360835
fixes RNBW-1140

for autoscrolling the height calculations we're totally f***ed, we needed to handle the current heights not what they we're previously, etc etc. The scrollToOffset function was also not working to begin with, i removed it and just call the function directly 

we now only scroll if the section doesnt fit on the screen currently, and if someone adds to the showcase

i QA'd this pretty hard, PoW show's 3 wallets with different wallet sections combinations

https://user-images.githubusercontent.com/29204161/135538087-c13836e2-a1a7-4957-a199-9cfd2572ba54.mp4

smol balances open: 

https://user-images.githubusercontent.com/29204161/135538127-ac30bada-da03-4cb6-8ffa-d4d83f7164a3.mp4

### CODE REVIEW

File: src/components/asset-list/RecyclerAssetList/index.tsx [hunk 1]
```diff
@@ -717,26 +712,97 @@ function RecyclerAssetList({
...
     });
+    const balancesHeight = balancesRows.length * CoinRowHeight;
+    //-3 for pixel perfection
+    const smallBalancesHeight =
+      ViewTypes.COIN_SMALL_BALANCES.calculateHeight({
+        isCoinListEdited: isCoinListEdited,
+        isOpen: openSmallBalances,
+        smallBalancesLength: smallBalances?.assets?.length || 0,
+      }) +
+      coinDividerHeight -
```
Comment (brunobar79) [inline src/components/asset-list/RecyclerAssetList/index.tsx:757]: Pretty sure the linter is complaining about the formatting here 😅

**Review by brunobar79 — APPROVED** [review]
Good stuff ser 🤝

**Review by osdnk — APPROVED** [review]
It's ok!

File: src/components/asset-list/RecyclerAssetList/index.tsx [hunk 2]
```diff
@@ -816,18 +842,16 @@ function RecyclerAssetList({
...
-        return data.item?.familyName === 'Showcase';
-      });
-
-      const layout = ref?.getLayout(familyIndex);
-      if (layout) {
-        const { y: startOfDesiredComponent } = layout;
-        scrollToOffset(startOfDesiredComponent - AssetListHeaderHeight, true);
-      }
+      const showcaseHeight = colleciblesStartHeight + AssetListHeaderHeight;
+      setTimeout(
```
Comment (osdnk) [inline src/components/asset-list/RecyclerAssetList/index.tsx:846]: why?

**Comment by osdnk** [comment]
another shitty library that makes our code shitty, but letz go

File: src/components/asset-list/RecyclerAssetList/index.tsx (hunk 2 above)
Comment (skylarbarrera) [reply src/components/asset-list/RecyclerAssetList/index.tsx:846]: wasnt working

**Pull request merged**